pkg/driver/nydus/nydus.go

```diff
@@ -178,7 +184,14 @@ func (d *Driver) Convert(ctx context.Context, provider accelcontent.Provider, so
 		return nil, err
 	}
 	if d.mergeManifest {
-		return d.makeManifestIndex(ctx, provider.ContentStore(), *image, *desc)
+		desc, err = d.makeManifestIndex(ctx, provider.ContentStore(), *image, *desc)
+		if err != nil {
+			return nil, err
+		}
+	}
+	desc.Annotations = map[string]string{
```

Comment:
Which we should append the annotation? The manifest index (aka manifest list) or the child manifest?

Reply:
if an artifact have index, add annotations to index layer, otherwise, just add to manifest.

Reply:
We should just add annotation to manifest layer